workflows/prognostic_c48_run/runtime/steppers/prescriber.py

```diff
@@ -162,45 +98,6 @@ def get_momentum_diagnostics(self, state, tendency) -> Diagnostics:
         return {}
 
 
-def _get_prescribed_ds(
-    dataset_key: str,
-    variables: Sequence[str],
-    timesteps: Optional[Sequence[cftime.DatetimeJulian]],
-    consolidated: bool = True,
-) -> Tuple[xr.Dataset, xr.DataArray]:
-    logger.info(f"Setting up dataset for state setting: {dataset_key}")
-    ds = _open_ds(dataset_key, consolidated)
-    ds = get_variables(ds, variables)
-    if timesteps is not None:
-        ds = _time_interpolate_data(ds, timesteps, variables)
-    time_coord = ds.coords["time"]
-    return ds.drop_vars(names="time").load(), time_coord
-
-
-def _time_interpolate_data(ds, timesteps, variables):
-    vars_interp_nearest = [var for var in variables if var in INTERPOLATE_NEAREST]
```

Comment:
Just to confirm, this functionality for discrete values is covered in the `time_interpolation_func`?

Reply:
Correct, this refactor means it no longer has to be implemented twice.